frontend/src/test/scala/bloop/bsp/BspCompileSpec.scala

```diff
@@ -349,6 +349,77 @@ class BspCompileSpec(
     }
   }
 
+  test("compile incrementally renamed file") {
+    TestUtil.withinWorkspace { workspace =>
+      def path(suffix: String) = s"/main/scala/scala/meta/internal/metals/Foo$suffix.scala"
+      object Sources {
+        val `Foo0.scala` =
+          s"""${path("0")}
+             |package scala.meta.internal.metals
+             |
+             |final class Foo0(conn: () => String) {
+             |  def foo(s: String): String = s
+             |}
+          """.stripMargin
+      }
+
+      val logger = new RecordingLogger(ansiCodesSupported = false)
+      val `A` = TestProject(workspace, "a", List(Sources.`Foo0.scala`))
+      val projects = List(`A`)
+      val fooFilePath = `A`.srcFor(path("0"))
+      writeFile(fooFilePath, Sources.`Foo0.scala`)
+
+      def assertCorrectArtifacts(compiledState: ManagedBspTestState, suffix: String): Unit = {
+
+        val buildProject = compiledState.toTestState.getProjectFor(`A`)
+        val externalClassesDirA =
+          compiledState.client.getUniqueClassesDirFor(buildProject, forceGeneration = true)
+
+        val classFilesAfterSuccess = takeDirectorySnapshot(externalClassesDirA)
+        assertEquals(
+          classFilesAfterSuccess.map(_.path.toString()),
+          List(s"/scala/meta/internal/metals/Foo$suffix.class")
+        )
+      }
+
+      loadBspState(workspace, projects, logger) { state =>
```

Comment:
I haven't been able to reproduce the issue in the test unfortunately, this seems somehow connected to multiple thing being done at the same time via metals and VS Code, but the change in BloopNameHashing is sound anyway.

And having this test won't hurt